### PR TITLE
CXP-1358: Send notif when an attribute is removed and is used by catalog mapping

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: upgrades/schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pim\Upgrade\Schema;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance extends AbstractMigration
```
Comment (guillaumefer-akeneo) [inline upgrades/schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance.php:10]: Version 8 ?

**Review by hogren — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingAttributesInProductMappingQuery.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        private readonly Connection $connection,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     * @throws \JsonException
+     */
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingAttributesInProductMappingQuery.php:26]: useless

File: components/catalogs/back/src/Infrastructure/Service/DispatchAttributeRemovedEvent.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Service/DispatchAttributeRemovedEvent.php:4]: rm blank line

File: upgrades/schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance.php (hunk 1 above)
Comment (masacc) [reply upgrades/schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance.php:10]: yes, no catalog feature in 7.0 so the jobs will be in 8.0

File: upgrades/test_schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance_Integration.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        parent::setUp();
+        $this->connection = $this->get('database_connection');
+    }
+
+    public function test_it_does_nothing_if_the_job_already_exists(): void
+    {
+        Assert::assertTrue($this->jobInstanceExists());
+        $this->reExecuteMigration(self::MIGRATION_NAME);
+        Assert::assertTrue($this->jobInstanceExists());
```
Comment (masacc) [inline upgrades/test_schema/Version_8_0_20230111140000_add_notify_on_attribute_removal_job_instance_Integration.php:34]: maybe you could test the count instead to be sure there is one and only one?

**Review by masacc — APPROVED** [review]

**Pull request merged**